src/main/java/com/google/impactdashboard/server/data_update/DataUpdater.java

```diff
@@ -80,40 +82,48 @@ public void updateDatabase() {
     List<ProjectIdentification> projects, String timeFrom, String timeTo) {
     return projects.parallelStream()
       .map(project -> {
-        ListLogEntriesPagedResponse response = logRetriever.listRecommendationLogs(
-          project.getProjectId(), timeFrom, timeTo);
-        List<LogEntry> entries = StreamSupport.stream(response.iterateAll().spliterator(), false)
-          .collect(Collectors.toList());
-        return recommendationRetriever.listRecommendations(
-          entries, project.getProjectId(), 
-          Recommendation.RecommenderType.IAM_BINDING, iamRetriever);
+        try {
+          ListLogEntriesPagedResponse response = logRetriever.listRecommendationLogs(
+              project.getProjectId(), timeFrom, timeTo);
+          List<LogEntry> entries = StreamSupport.stream(response.iterateAll().spliterator(), false)
+              .collect(Collectors.toList());
+          return recommendationRetriever.listRecommendations(
+              entries, project.getProjectId(),
+              Recommendation.RecommenderType.IAM_BINDING, iamRetriever);
+        } catch (Exception e) {
+          return new ArrayList<Recommendation>();
```

Comment:
What could have caused the exception? Is it safe to ignore?

Reply:
The exception that is the issue here is an issue where we found out that some projects can block access to cloud logging and we had no idea how to stop this because it would crash the program as soon as it ran any cloud logging request.